src/Microsoft.AspNetCore.Sockets/ConnectionManager.cs

```diff
@@ -82,7 +83,10 @@ private static string MakeNewConnectionId()
 
         private static void Scan(object state)
         {
-            ((ConnectionManager)state).Scan();
+            if (!Debugger.IsAttached)
```

Comment:
Yank this before you merge. Though @davidfowl , I'm beginning to think this would be useful to add to DEBUG builds... it's a pain to debug when connections keep getting cut out from under you ;)